pkg/supportbundle/spec.go

```diff
@@ -844,79 +696,22 @@ func getImageAndSecret(ctx context.Context, clientset kubernetes.Interface) (ima
 	return imageName, pullSecret, nil
 }
 
-func makeGoldpingerCollectors() []*troubleshootv1beta2.Collect {
-	collectors := []*troubleshootv1beta2.Collect{}
-
-	collectors = append(collectors, &troubleshootv1beta2.Collect{
-		Exec: &troubleshootv1beta2.Exec{
-			CollectorMeta: troubleshootv1beta2.CollectorMeta{
-				CollectorName: "goldpinger-statistics",
-			},
-			Name:          "kots/goldpinger",
-			Selector:      []string{"app=kotsadm"},
-			ContainerName: "kotsadm",
-			Command:       []string{"curl"},
-			Args:          []string{"http://goldpinger.kurl.svc.cluster.local:80/check_all"},
-			Timeout:       "10s",
-		},
-	})
-
-	return collectors
-}
-
-func makeGoldpingerAnalyzers() []*troubleshootv1beta2.Analyze {
-	analyzers := []*troubleshootv1beta2.Analyze{}
-
-	analyzers = append(analyzers, &troubleshootv1beta2.Analyze{
-		TextAnalyze: &troubleshootv1beta2.TextAnalyze{
-			AnalyzeMeta: troubleshootv1beta2.AnalyzeMeta{
-				CheckName: "Inter-pod Networking",
-			},
-			FileName:    "kots/goldpinger/*/kotsadm-*/goldpinger-statistics-stdout.txt",
-			RegexGroups: `"OK": ?(?P<OK>\w+)`,
-			Outcomes: []*troubleshootv1beta2.Outcome{
-				{
-					Fail: &troubleshootv1beta2.SingleOutcome{
-						When:    "OK = false",
-						Message: "Some nodes have pod communication issues",
-					},
-				},
-				{
-					Pass: &troubleshootv1beta2.SingleOutcome{
-						Message: "Goldpinger can communicate properly",
-					},
-				},
-			},
-		},
-	})
-
-	return analyzers
-}
-
-func makeLonghornCollectors() []*troubleshootv1beta2.Collect {
-	collectors := []*troubleshootv1beta2.Collect{}
+func populateImages(supportBundle *troubleshootv1beta2.SupportBundle, imageName string, pullSecret *troubleshootv1beta2.ImagePullSecrets) *troubleshootv1beta2.SupportBundle {
+	next := supportBundle.DeepCopy()
 
-	collectors = append(collectors, &troubleshootv1beta2.Collect{
-		Longhorn: &troubleshootv1beta2.Longhorn{
-			CollectorMeta: troubleshootv1beta2.CollectorMeta{
-				CollectorName: "longhorn",
-			},
-		},
-	})
-
-	return collectors
-}
-
-func makeLonghornAnalyzers() []*troubleshootv1beta2.Analyze {
-	analyzers := []*troubleshootv1beta2.Analyze{}
-
-	analyzers = append(analyzers, &troubleshootv1beta2.Analyze{
-		Longhorn: &troubleshootv1beta2.LonghornAnalyze{
-			AnalyzeMeta: troubleshootv1beta2.AnalyzeMeta{
-				CheckName: "longhorn",
-			},
-		},
-	})
+	collects := []*troubleshootv1beta2.Collect{}
+	for _, collect := range next.Spec.Collectors {
+		if collect.Collectd != nil && collect.Collectd.Image == "alpine" { // TODO: is this too strong of an assumption?
```

Comment:
I'm assuming the intention here is to override images in the embedded spec only.  If so, how about calling this before adding any other collectors.

Reply:
good call. will make that change